rest/middleware/cors.go

```diff
@@ -13,25 +13,32 @@ import (
 // a MiddlewareError if the request should be terminated.
 func NewCORSMiddleware(originWhitelist []string) rest.Middleware {
 	return func(w http.ResponseWriter, r *http.Request) *rest.MiddlewareError {
+		origin := r.Header.Get("Origin")
 		originMatch := false
-		if origin := r.Header.Get("Origin"); checkOrigin(origin, originWhitelist) {
+		if checkOrigin(origin, originWhitelist) {
```

Comment:
Should we add a check for the origin header being missing here, as opposed to doing the check unnecessarily?